### PR TITLE
Add benchmarks to pipeline

### PR DESCRIPTION
**Pull request by alvicsam**
PR adds manual benchmarks job. It's divided to 2 jobs: `benchmarks-build` which builds binary on a faster ci runners and passes to `benchmarks` job which execute benchmarks. 
The `benchmarks` job runs script `benchmarks-ci.sh` for statemine, statemint and westmint and creates PR on github with updated weights (example: https://github.com/paritytech/cumulus/pull/1138)

Part of https://github.com/paritytech/ci_cd/issues/260
Closes https://github.com/paritytech/cumulus/issues/631

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -8,8 +8,19 @@
 stages:
   - test
   - publish
+  - benchmarks-build
+  - benchmarks
```
Comment (TriplEight) [inline .gitlab-ci.yml:12]:
```suggestion
  - benchmarks-run
```

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -172,3 +193,47 @@ publish-s3:
...
+  <<:                              *docker-env
+  <<:                              *collect-artifacts
+  <<:                              *benchmarks-refs
+  script:
+    - time cargo build --profile production --locked --features runtime-benchmarks
+    - mkdir artifacts
+    - cp $CARGO_TARGET_DIR/production/polkadot-collator ./artifacts/
+
+benchmarks-statemine:
+  stage:                           benchmarks
```
Comment (TriplEight) [inline .gitlab-ci.yml:210]:
```suggestion
  stage:                           benchmarks-run
```

**Review by TriplEight — CHANGES_REQUESTED** [review]

**Review by TriplEight — APPROVED** [review]

**Comment by alvicsam** [comment]
There is an additional condition that should be implemented (after discussing it with @NachoPal in element):

Weights should be pushed to the same branch where the binary for benchmarks was compiled.

Example:

1. Select branch release-parachains-vX.Y.Z -> Click Run benchmarks button
2. The binary is built, benchmarks are run.
3. When the new weights are ready, a PR is open against release-parachains-vX.Y.Z
4. We review the PR and merge it to release-parachains-vX.Y.Z

**Review by NachoPal — CHANGES_REQUESTED** [review]
Thanks @alvicsam . I realised that we'll need an input to select the branch we want to use for building the binary, running the benchmarks and opening a PR with the new weights.

The flow would be as follows:
1. Select branch release-parachains-vX.Y.Z -> Click Run benchmarks button
2. The binary is built, benchmarks are run.
3. When the new weights are ready, a PR is open against release-parachains-vX.Y.Z
4. We review the PR and merge it to release-parachains-vX.Y.Z

**Comment by NachoPal** [comment]
What Benchmarking machine is used to run the benchmarks?

**Comment by alvicsam** [comment]
`bm4`, `bm5`, `bm6`

**Pull request merged**